web_monitoring/diffing_server.py

```diff
@@ -75,15 +75,21 @@ def __init__(self, url, body, headers):
 access_control_allow_origin_header = \
     os.environ.get('ACCESS_CONTROL_ALLOW_ORIGIN_HEADER')
 
-
 class BaseHandler(tornado.web.RequestHandler):
 
+    _ALLOWED_ORIGINS = None
+
     def set_default_headers(self):
         if access_control_allow_origin_header is not None:
-            self.set_header("Access-Control-Allow-Origin",
-                            access_control_allow_origin_header)
-            self.set_header("Access-Control-Allow-Credentials", "true")
-            self.set_header("Access-Control-Allow-Headers", "x-requested-with")
+            if not self._ALLOWED_ORIGINS:
+                self._ALLOWED_ORIGINS = set([origin.strip() for origin in
```

Comment:
In Tornado, `RequestHandler` instances get constructed for each request, so this caching doesn’t really do any good. You should cache it in the application’s settings dict. You can get that either from `RequestHandler.application.settings` or just `RequestHandler.settings`:

```py
if not self.settings['allowed_origins']:
    self.settings['allowed_origins'] = set(...)
```

More in the docs:
- http://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.settings
- http://www.tornadoweb.org/en/stable/web.html#tornado.web.Application.settings

Reply:
If you do `if not self.settings['allowed_origins']:` and `allowed_origins` is not set, we'll have an error.
Its better to do: `if 'allowed_origins' not in self.settings:`

Reply:
Thank you for explaining this, I'm fixing it now.